pkg/provisioner/provisioner.go

```diff
@@ -0,0 +1,273 @@
+package provisioner
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+
+	"github.com/iver-wharf/wharf-core/pkg/logger"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+var log = logger.NewScoped("PROVISIONER")
+
+var podInitCloneArgs = []string{"git", "clone"}
+var podContainerListArgs = []string{"/bin/sh", "-c", "ls -alh"}
+
+// Provisioner is an interface declaring what methods are required
+// for a provisioner.
+type Provisioner interface {
+	Serve(ctx context.Context) error
+	ListPods(ctx context.Context) error
+	DeletePod(ctx context.Context, UID string) error
```

Comment:
Rename ListPods->ListWorkers & DeletePod->DeleteWorker, because that's really what the provisioner is doing. Kubernetes pods are just an implementation detail.

Also, rename parameter `UID` -> `id` because 1. parameters should start with lowercase and 2. because UIDs are an implementation detail for Kubernetes. Could even rename it to `workerID` to be more explicit.